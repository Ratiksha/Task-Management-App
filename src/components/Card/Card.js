import React, { Component } from 'react';
import './Card.scss';
import Modal from '../Modal/Modal'
import AddTaskForm from '../AddTaskForm/AddTaskForm'
import DeleteForm from '../DeleteForm/DeleteForm'
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashAlt, faPencilAlt } from "@fortawesome/free-solid-svg-icons";

export default class Card extends Component{
    constructor(props){
        super(props);
        this.state = {
            showModal: false,
            showEditForm: false,
            showDeleteForm: false
        }
        this.handleKeyPress = this.handleKeyPress.bind(this);
    }

    componentDidMount(){
        document.addEventListener('keydown', this.handleKeyPress);
    }

    handleKeyPress(event) {
        if (event.keyCode === 13) {
            document.activeElement.click();
        }
      }

    showModal = (formType) => {
        this.toggleModal();
        if(formType === 'edit'){
            this.setState({ 
                showEditForm: true,
                showDeleteForm: false
            });
        }
        else if(formType === 'delete'){
            this.setState({ 
                showDeleteForm: true,
                showEditForm: false
            });
        }
    }
    
    hideModal = () => {
        this.toggleModal();
    }

    toggleModal = () => {
        this.setState({ 
            showModal: !this.state.showModal,
            showDeleteForm: false,
            showEditForm: false
         });
    }

    render() {
        const { card, lists, editTask, DeleteTask, listId, onTaskDragStart } = this.props
        return (
            <div className="card" draggable="true" id={[card && card.taskId]} onDragStart={onTaskDragStart} onClick={(event)=> event.stopPropagation()}>
                {card && card.taskName}
                <div className="button-group">
                    <button id="card-edit" onClick={() => this.showModal('edit')}><FontAwesomeIcon icon={faPencilAlt}/></button>
                    <button id="card-delete" onClick={() => this.showModal('delete')}><FontAwesomeIcon icon={faTrashAlt}/></button>
                </div>
                <Modal show={this.state.showModal} handleClose={this.hideModal}>
                {this.state.showEditForm && <AddTaskForm lists={lists} listId={listId} name={card && card.taskName} taskId={card && card.taskId} editTask={editTask} hideModal={this.hideModal}/>}
                {this.state.showDeleteForm && <DeleteForm listId={listId} name={card && card.taskName} taskId={card && card.taskId} DeleteTask={DeleteTask} hideModal={this.hideModal}/>}
                </Modal>
            </div>
        )
    }
}