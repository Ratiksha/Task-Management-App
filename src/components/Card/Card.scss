@import '../variables.scss';

.card{
    background: $white;
    border-left: 3px solid var(--color);;
    margin-bottom: 8px;
    padding: 30px 10px;
    word-break: break-all;
    cursor: pointer;
    position: relative;
    box-shadow: -1px 2px 4px 2px #75757514;
    .button-group{
        position: absolute;
        top: 10px;
        right: 10px;
        button{
            background-color: transparent;
            padding: 0;
        }
        svg{
            color: #00000052;
            margin: 5px;
            font-size: 10px;
        }
    }
}