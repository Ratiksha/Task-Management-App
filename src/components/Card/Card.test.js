import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

let wrapped = shallow(<Card />);
let editType = 'edit';
let deleteType = 'delete';
describe('Card', () => {
  it('should render the Card Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('When edit button is clicked', () => {
    const editButton = wrapped.find('#card-edit');
    editButton.simulate('click');
    expect(editType).toBe('edit');
  });

  it('When delete button is clicked', () => {
    const deleteButton = wrapped.find('#card-delete');
    deleteButton.simulate('click');
    expect(deleteType).toBe('delete');
  });
});