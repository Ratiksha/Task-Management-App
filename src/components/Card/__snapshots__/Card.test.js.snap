// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render the Card Component correctly 1`] = `ShallowWrapper {}`;
